androidApp/src/main/java/com/mbta/tid/mbta_app/android/map/MapLayerManager.kt

```diff
@@ -28,51 +32,62 @@ class MapLayerManager(val map: MapboxMap, context: Context) {
         }
     }
 
-    fun addSource(source: GeoJsonSource) {
-        this.map.addSource(source)
+    @AnyThread
+    suspend fun addSource(source: GeoJsonSource) {
+        withContext(Dispatchers.Main) { map.addSource(source) }
     }
 
-    fun addLayers(colorPalette: ColorPalette) {
+    @AnyThread
+    suspend fun addLayers(colorPalette: ColorPalette) {
         val layers: List<MapboxLayer> =
-            RouteLayerGenerator.createAllRouteLayers(colorPalette).map { it.toMapbox() } +
-                StopLayerGenerator.createStopLayers(colorPalette).map { it.toMapbox() }
-        for (layer in layers) {
-            if (map.styleLayerExists(checkNotNull(layer.layerId))) {
-                // Skip attempting to add layer if it already exists
-                continue
+            withContext(Dispatchers.Default) {
+                RouteLayerGenerator.createAllRouteLayers(colorPalette).map { it.toMapbox() } +
+                    StopLayerGenerator.createStopLayers(colorPalette).map { it.toMapbox() }
             }
-            if (map.styleLayerExists("puck")) {
-                map.addLayerBelow(layer, below = "puck")
-            } else {
-                map.addLayer(layer)
+        withContext(Dispatchers.Main) {
+            for (layer in layers) {
+                if (map.styleLayerExists(checkNotNull(layer.layerId))) {
+                    // Skip attempting to add layer if it already exists
+                    continue
+                }
+                if (map.styleLayerExists("puck")) {
+                    map.addLayerBelow(layer, below = "puck")
+                } else {
+                    map.addLayer(layer)
+                }
             }
         }
     }
 
+    @MainThread
```

Comment:
If this were `@AnyThread`, then we would actually get an error from the Android thread annotation checker, because `styleLayerExists` and `moveStyleLayer` have `@MainThread`. Unfortunately, it looks like callbacks in general just kinda break that checking process entirely, because nothing gets actually checked inside `withContext() {}` even if it's on the wrong dispatcher, so more comprehensive use of these annotations will not gain us much, and even this is unlikely to actually catch an error for us. If these annotations are confusing (which might be the case, especially by contrast with iOS where `@MainActor` moves its contents to the main thread instead of merely documenting that they should be called there), I can take them back out.

Reply:
Oh interesting point about the callbacks & possible confusion with iOS annotation. If they won't catch many errors & are a possible source of confusion, I'd be in favor of leaving them out